src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -119,5 +142,61 @@ private CodeAction TryCreateCodeAction(Document document, AddImportFixData fixDa
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return new AddImportCodeFixAllProvider();
+        }
+
+        internal async Task<Document> FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var getFixesTasks = diagnostics
+                .GroupBy(d => d.Location.SourceSpan)
+                .Select(g => GetFixesForDiagnosticsAsync(document, g.Key, g.AsImmutable(), true, false, cancellationToken));
+
+            var usableFixes = (await Task.WhenAll(getFixesTasks).ConfigureAwait(false))
```

Comment:
I dont think getFixesTasks will return deterministic list, so what is the reverse is for? list is not sorted list?